File: LR_2/task2.java
// Двусвязный список элемент структуры
struct Element {
    int value;
    Element *prev;
    Element *next;
};

// Очередь строк
std::queue<std::string> namesQueue;
namesQueue.push("Alice");
namesQueue.push("Bob");
namesQueue.push("Eve");

// Дек
std::deque<int> numbers = {10, 20, 30};

// Приоритетная очередь для int
std::priority_queue<int> intPQ;
intPQ.push(100);
intPQ.push(50);
intPQ.push(75);

// Приоритетная очередь для пользовательских структур
struct Job {
    std::string name;
    int priority;
};

struct CompareJobs {
    bool operator()(Job const& a, Job const& b) {
        return a.priority > b.priority; // Минимальная очередь по приоритету
    }
};

std::priority_queue<Job, std::vector<Job>, CompareJobs> jobPQ;
jobPQ.push({"Cleaning", 3});
jobPQ.push({"Installing", 1});
jobPQ.push({"Repair", 2});

File: LR_2/task2.py
# Пример создания вложенного списка (группы студентов)
teams = [['Alice', 'Bob'], ['Eve', 'Charlie'], ['Dave', 'Frank']]

# Пример создания очереди
from queue import Queue
line = Queue()
line.put(5)
line.put(10)
line.put(15)

# Пример реализации дека
from collections import deque
job_queue = deque()
job_queue.append("job1")
job_queue.append("job2")
job_queue.append("job3")

# Пример приоритетной очереди
from queue import PriorityQueue
priority_tasks = PriorityQueue()
priority_tasks.put((3, "low priority"))
priority_tasks.put((1, "urgent"))
priority_tasks.put((2, "normal"))

# Пример приоритетной очереди с бинарной кучей
import heapq
clients = []
heapq.heappush(clients, (5, "John"))
heapq.heappush(clients, (3, "Anna"))
heapq.heappush(clients, (7, "Zara"))
heapq.heappush(clients, (1, "Mia"))
